mycore-base/src/main/java/org/mycore/frontend/cli/MCRObjectCommands.java

```diff
@@ -1060,6 +1061,175 @@ public static void checkDerivatesInObjects(String baseId) {
         LOGGER.info("Check done for {} entries", Integer.toString(counter));
     }
 
+    /**
+     * Checks objects of the specified base ID for validity against their specified schemas.
+     * 
+     * @param baseId
+     *            the base part of a MCRObjectID e.g. DocPortal_document
+     */
+    @MCRCommand(
+        syntax = "validate objects of base {0}",
+        help = "Validates all objects of base {0} against their specified schema.",
+        order = 401)
+    public static void validateObjectsOfBase(String baseID) {
+        if (baseID == null || baseID.length() == 0) {
+            LOGGER.error("Base ID of objects required to check their schema validity.");
+            return;
+        }
+        MCRXMLMetadataManager mgr = MCRXMLMetadataManager.instance();
+        List<String> idList;
+        try {
+            idList = mgr.listIDsForBase(baseID);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            LOGGER.error("Requested base ID {} not found in store.", baseID);
+            return;
+        }
+        int counter = 0;
+        int maxresults = idList.size();
+        if (maxresults == 0) {
+            LOGGER.warn("No IDs returned for base {}, nothing to check.", baseID);
+            return;
+        }
+        for (String objID : idList) {
+            ++counter;
+            LOGGER.info("Processing dataset {} of {} with ID: {}", counter, maxresults, objID);
+            validateObject(objID);
+        }
+        LOGGER.info("Check done for {} entries", counter);
+    }
+
+    /**
+     * Checks objects of the specified type for validity against their specified schemas.
+     * 
+     * @param type
+     *            the type of a MCRObjectID e.g. document
+     */
+    @MCRCommand(
+        syntax = "validate objects of type {0}",
+        help = "Validates all object of type {0} against their specified schema.",
+        order = 402)
+    public static void validateObjectsOfType(String type) {
```

Comment:
If you dont use anything returned from the validateObject method, then its better to just return a list of commands which should be executed. MyCoRe will then execute these commands. The command org.mycore.frontend.cli.MCRObjectCommands#deleteAllObjects is a good example. 
 
You could also let the validateObject method throw an exception. MyCoRe puts all failed commands to a .txt file in the home directory. You then have a list of a failed objects. You can fix them and then run the command list with "process <txtfile>".